doc/design/diagrams/iceoryx_components_diagram_v3_0_0.puml

```diff
@@ -27,26 +28,42 @@ skinparam frame {
 
 header Eclipse iceoryx component overview
 
+' Platform Library containing platform abstraction with quality level 1
+package iceoryx_platform  {
+   frame "platform library" {
+      component "Platform abstraction" <<module>>
+   }
+}
+
 ' HOOFS: Helpful Objects Optimised For Safety
-' Library containing modern C++ STL constructs
+' Library containing modern C++ STL constructs with quality level 1+
 package iceoryx_hoofs  {
    frame "hoofs library" {
-      component concurrent <<namespace>>
-      component cxx <<namespace>> {
-         component newtype <<namespace>>
-         component set <<namespace>>
-         component algorithm <<namespace>>
-      }
-      component DesignPattern <<namespace>>
-      component errors <<namespace>>
-      component log <<namespace>> {
-         component ffbb <<namespace>>
-      }
-      component units <<namespace>> {
-         component duration_literals <<namespace>>
-
-      }
-      component posix <<namespace>>
+
+      component "Memory & lifetime management" <<module>>
+      component Containers <<module>>
+      component "Common vocabulary types" <<module>>
+      component "Filesystem & OS" <<module>>
+      component Functional <<module>>
+      component Utility <<module>>
+      component Strings <<module>>
+      component Types <<module>>
+      component Algorithms <<module>>
+      component "Generalized design patterns" <<module>>
+      component "Error handling & logging" <<module>>
+      component Time <<module>>
+      component "Queues & communication" <<module>>
+      component "Threads & Sychronisation" <<module>>
```

Comment:
This does not make sense. It does not resemble anything we have in our codebase. It might be a logical ordering but it does not tell me anything about our structure. Where do I find `Common vocabulary types` or `Queues & communication`? It's not there. This does not help me to get a quick overview but just confuses me once I look at the codebase.

It is okay to have a list with classes under these categories (e.g. in the hoofs README) but these categories do not reflect the current architecture of hoofs.

Reply:
@elBoberido @elfenpiff 

I think there was a misunderstanding here. As [discussed yesterday](https://github.com/eclipse-iceoryx/iceoryx/issues/1593#issuecomment-1252488853), `iceoryx_components_diagram_v3_0_0.puml` shall reflect the planned user-view for `v3.0`. It shall give users an easy way to grasp what's inside `iceoryx_hoofs`. It's a great idea to update the `Readme.md` alongside these changes to make the mapping clearer. I've updated it accordingly. How the classes are structured for developers inside the repository is a different story. We can discuss that as part of #1391

Reply:
@elfenpiff @elBoberido I renamed `Types` to `Meta` as suggested [here](https://github.com/eclipse-iceoryx/iceoryx/issues/1593#issuecomment-1253451255). Our internal structure does not necessarily has to be exactly the same as the user-view which is reflected in this diagram. Of course it would be good if it's not too different.

As discussed, internally the internal structure would look like [this](https://github.com/eclipse-iceoryx/iceoryx/issues/1593#issuecomment-1258266337) with these subfolders:

```
iceoryx_hoofs
|- container
|- vocabulary
|- functional
|- etc.
|- concurrent
  |- container
  |- queues
  |- etc.
|- posix
  |- sync
  |- memory
  |- etc.
```

Reply:
@mossmaurice Just as a fun fact what `etc` means. Originally it meant `etcetera` since the unix guys didn't find a better word how to name the folder where all the remaining stuff, which didn't fit into any category, is placed.

But today `etc` stands for `Editable Text Configurations`.

So with this being said I think that `etc` does not fit at all. We could also create a component `stuff` or `utility` or `widgets` :smile:

Yeah ... finding names is hard :disappointed:

Reply:
@elfenpiff Hehe, didn't know that :grin: Not sure if you're kidding here, but `etc.` in the above post stands for any other module name :stuck_out_tongue: